### PR TITLE
chgres_cube: option to build without nemsio, sigio and sfcio libraries

### LINKED ISSUE #925: Add option to build chgres_cube without nemsio and sigio input file support
Add option to build `chgres_cube` without support for nemsio or sigio/sfcio input data. 

It is suspected that most users only are interested in coldstarting recent cases (within the past few years).  Also, users who run `chgres_cube` for simple cases on their own laptop don't require the option that uses GFS nemsio data as input, and the nemsio library can be burdensome to port.

This will be an option. The build will be modified in a way that nemsio and sigio data can still be used. However, the standard build will not support these data. After a period of six months or so, if users do not complain, then support for these older input data will be completely remove from `chgres_cube`.

### PR DESCRIPTION
**Pull request by GeorgeGayno-NOAA**
## DESCRIPTION OF CHANGES: 
Add logic to build `chgres_cube` without the `nemsio`, `sigio` and `sfcio` libraries. The `nemsio`
library can be difficult for users to build on their workstations. These libraries support
older input data (GFS v14 and earlier).  The default will be to build `chgres_cube` without
these libraries. How the program is built is determined by the `CHGRES_ALL` variable in
the main `CMakeLists.txt` file (set to 'ON' for full build).

The `chgres_cube` unit tests `ftst_read_sfc_gfs_nemsio`, `ftst_read_sfc_nemsio` and
`ftst_read_nst_nemsio` are retained and run only when `CHGRES_ALL` is 'ON'.

The `chgres_cube` regression tests `c96.fv3.nemsio`, `c96.gfs.sigio` and `c96.gfs.nemsio`
are retained, but not run by default. They may be optionally run on Hera when 
XTRA_TESTS is set to "TRUE" in its driver script.

The `gdas_init` utility was updated to use the v15 GDAS tarball for coldstarting GFS
free forecasts. The v15 GFS tarball contains `nemsio` data. The GDAS tarball contains
warm restart files in netcdf. The logic was updated to stop if users choose v14 or
earlier data as input.

## TESTS CONDUCTED: 
If there are changes to the build or source code, the tests below must be conducted. Contact a repository manager if you need assistance.

- [x] Compile branch on all Tier 1 machines using Intel (Orion, Jet, Hera, Hercules and WCOSS2). Done using 3e58ed3.
- [x] Compile branch on Hera using GNU. Done using 3e58ed3.
- [x] Compile branch in 'Debug' mode on WCOSS2. Done on Dogwood using 3e58ed3.
- [x] Run unit tests locally on any Tier 1 machine. Done on Hera using 3e58ed3 using both builds of chgres. All tests passed.
- [x] Run chgres_cube consistency tests locally on all Tier 1 machines and 's4'. Done using 3e58ed3. On Hera, both builds of chgres were tested. All tests passed.

Other tests:

- The branch was successfully compiled on:
  -  's4', (https://github.com/ufs-community/UFS_UTILS/pull/928#issuecomment-2056616884) 
  -  the 'cloud' (https://github.com/ufs-community/UFS_UTILS/pull/928#issuecomment-2057264624)
  -  a laptop (https://github.com/ufs-community/UFS_UTILS/issues/925#issuecomment-2025498142)

- The `gdas_init` utility was tested (3e58ed3) on a v15 test case (Jan 1, 2020, 06z) for both GFS and GDAS. It worked correctly on WCOSS2, Jet, Hera and 's4' (https://github.com/ufs-community/UFS_UTILS/pull/928#discussion_r1568951710)

## DEPENDENCIES:
None.

## DOCUMENTATION:
The 'readthedocs' was updated: https://georgegayno-noaaufs-utils.readthedocs.io/en/feature-chgres_simple/ufs_utils.html#configure-for-your-experiment

The Doxygen was created on Hera using 3e58ed3. There were no warnings. See here: https://georgegayno-noaa.github.io/UFS_UTILS/chgres_cube/index.html

## ISSUE: 
Fixes #925.

### CODE REVIEW

**Comment by GeorgeGayno-NOAA** [comment]
@DavidHuber-NOAA - I don't have access to 's4'. Can you please compile my branch and run the `chgres_cube` regression tests? Thanks.

**Comment by GeorgeGayno-NOAA** [comment]
@HenryWinterbottom-NOAA - I don't have access to the 'cloud'. Could you please compile my branch on the 'cloud'? Thanks.

**Comment by GeorgeGayno-NOAA** [comment]
@DusanJovic-NOAA - I also removed the dependency on the `sigio` and `sfcio` libraries. If you just build `chgres_cube`, you no longer need `nemsio`, `sigio` and `sfcio`. Can you please test my branch again on your laptop? Thanks.

**Comment by HenryRWinterbottom** [comment]
@GeorgeGayno-NOAA yes.

I am on leave at the moment, but I will take care of it Friday or ASAP next week.

**Comment by GeorgeGayno-NOAA** [comment]
@aerorahul and @KateFriedman-NOAA - this PR will affect users of gdas_init. They will no longer be able to use 'old' input data. You know your users better than I do, so let me know if this will cause them grief.

**Comment by KateFriedman-NOAA** [comment]
> @aerorahul and @KateFriedman-NOAA - this PR will affect users of gdas_init. They will no longer be able to use 'old' input data. You know your users better than I do, so let me know if this will cause them grief.

From your changeset I see:
```
does NOT support experiments prior to June 12, 2019, or if CDUMP is 'gfs', experiments prior to March 21, 2021
```

The June 2019 date corresponds with the GFSv15 implementation and the March 2021 date corresponds with GFSv16 implementation into ops. So to confirm, nothing from ops pre-v15 will work going into chgres_cube now (gdas or gfs) and nothing from ops pre-v16 for CDUMP=gfs but v15 gdas works still?

Does this also include the pre-implementation parallel outputs for both implementations or just the ops outputs?

File: tests/chgres_cube/ftst_program_setup.F90 [hunk 1]
```diff
@@ -19,30 +19,27 @@ program ftst_program_setup
...
-  if (tracers(1) .ne. "sphum" .or. tracers(2) .ne. "liq_wat" .or. tracers(3) .ne. "o3mr" .or. &
-       tracers(4) .ne. "ice_wat" .or. tracers(5) .ne. "rainwat" .or. tracers(6) .ne. "snowwat" .or. &
-       tracers(7) .ne. "graupel") stop 22
+  if (cycle_mon .ne. 10 .or. cycle_day .ne. 3 .or. cycle_hour .ne. 0) stop 3
+  if (.not. convert_atm .or. .not. convert_sfc .or. .not. convert_nst) stop 4
+  if (regional .ne. 0 .or. halo_bndy .ne. 0 .or. halo_blend .ne. 0) stop 5
+  if (trim(mosaic_file_target_grid) .ne. "/scratch1/NCEPDEV/da/George.Gayno/noscrub/reg_tests/chgres_cube/fix/C192/C192_mosaic.nc") stop 6
+  if (trim(fix_dir_target_grid) .ne. "/scratch1/NCEPDEV/da/George.Gayno/noscrub/reg_tests/chgres_cube/fix/C192/fix_sfc") stop 7
+  if (trim(orog_dir_target_grid) .ne. "/scratch1/NCEPDEV/da/George.Gayno/noscrub/reg_tests/chgres_cube/fix/C192/") stop 8
+  if (trim(vcoord_file_target_grid) .ne. "/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/UFS_UTILS/reg_tests/chgres_cube/../../fix/fix_am/global_hyblev.l64.txt") stop 9
```
Comment (KateFriedman-NOAA) [inline tests/chgres_cube/ftst_program_setup.F90:28]: I see you had hardcoded personal account paths before and still have them here now. So this code only works for testing on Hera?

File: util/gdas_init/config [hunk 2]
```diff
@@ -91,13 +95,31 @@ else
...
+#   gfs_ver=v14
+    set +x
+    echo FATAL ERROR: V14 DATA NOT SUPPORTED.
+    echo CONTACT A REPOSITORY MANAGER FOR ASSISTANCE.
+    exit 2
   elif [ $yy$mm$dd$hh -lt 2021032100 ]; then
     gfs_ver=v15
+    if [ "$CDUMP" = 'gfs' ];then
+    set +x
+      echo FATAL ERROR: V15 DATA NOT SUPPORTED.
```
Comment (KateFriedman-NOAA) [inline util/gdas_init/config]: Suggest putting "" around the echo statements in here.

**Review by KateFriedman-NOAA — COMMENTED** [review]
I can't comment on the code changes but I acknowledge the update that means pre-v15 GFS and v15 GFS will no longer be supported by chgres_cube.

Left a few comments to address.

**Comment by DavidHuber-NOAA** [comment]
> @DavidHuber-NOAA - I don't have access to 's4'. Can you please compile my branch and run the `chgres_cube` regression tests? Thanks.

@GeorgeGayno-NOAA All tests passed on S4.

File: tests/chgres_cube/ftst_program_setup.F90 (hunk 1 above)
Comment (GeorgeGayno-NOAA) [reply tests/chgres_cube/ftst_program_setup.F90:28]: This is just a unit test that checks the read of the program namelist. It uses a dummy namelist. The paths are not used.

**Comment by GeorgeGayno-NOAA** [comment]
> > @aerorahul and @KateFriedman-NOAA - this PR will affect users of gdas_init. They will no longer be able to use 'old' input data. You know your users better than I do, so let me know if this will cause them grief.
> 
> From your changeset I see:
> 
> ```
> does NOT support experiments prior to June 12, 2019, or if CDUMP is 'gfs', experiments prior to March 21, 2021
> ```
> 
> The June 2019 date corresponds with the GFSv15 implementation and the March 2021 date corresponds with GFSv16 implementation into ops. So to confirm, nothing from ops pre-v15 will work going into chgres_cube now (gdas or gfs) and nothing from ops pre-v16 for CDUMP=gfs but v15 gdas works still?
> 
Correct. For v15, the GFS tarballs contain nemsio data, which won't be formally supported. The GDAS tarballs contain the tiled restart files in netcdf, which will be supported.  (Could we coldstart GFS forecasts with the GDAS tarballs?)

> Does this also include the pre-implementation parallel outputs for both implementations or just the ops outputs?

Do you mean coldstarts with the v16 retrospective parallel data? That data will still be supported.

**Comment by KateFriedman-NOAA** [comment]
> The GDAS tarballs contain the tiled restart files in netcdf, which will be supported. (Could we coldstart GFS forecasts with the GDAS tarballs?)

I don't see why not. The restart filenames are agnostic to `RUN`, it's just the tarball folder structure that says `gdas`. Would you be able to update the HPSS pull scripts for `gdas_init` to pull from the `gdas` tarballs for both `gdas` and `gfs` for v15?

> Do you mean coldstarts with the v16 retrospective parallel data? That data will still be supported.

Great! The warm starts from those retros.

File: util/gdas_init/config (hunk 2 above)
Comment (GeorgeGayno-NOAA) [reply util/gdas_init/config]: Done at 80a3f3f.

**Comment by HenryRWinterbottom** [comment]
> @HenryWinterbottom-NOAA - I don't have access to the 'cloud'. Could you please compile my branch on the 'cloud'? Thanks.

@GeorgeGayno-NOAA This branch builds against my stack on AWS PW.

**Comment by GeorgeGayno-NOAA** [comment]
The scripts were updated at 38683dc to pull/use the GDAS v15 tarball (which is netcdf) for both CDUMPs ('gfs' and 'gdas'). 

The 'gfs' and 'gdas' coldstart files were produced and placed in the proper directories:

```
/scratch2/NCEPDEV/stmp1/George.Gayno/gdas.init.branch/output/gfs.20200101/06/model_data/atmos/input $ ls
gfs_ctrl.nc        gfs_data.tile2.nc  gfs_data.tile4.nc  gfs_data.tile6.nc  sfc_data.tile2.nc  sfc_data.tile4.nc  sfc_data.tile6.nc
gfs_data.tile1.nc  gfs_data.tile3.nc  gfs_data.tile5.nc  sfc_data.tile1.nc  sfc_data.tile3.nc  sfc_data.tile5.nc
```

```
/scratch2/NCEPDEV/stmp1/George.Gayno/gdas.init.branch/output/gdas.20200101/06/model_data/atmos/input $ ls
gfs_ctrl.nc        gfs_data.tile2.nc  gfs_data.tile4.nc  gfs_data.tile6.nc  sfc_data.tile2.nc  sfc_data.tile4.nc  sfc_data.tile6.nc
gfs_data.tile1.nc  gfs_data.tile3.nc  gfs_data.tile5.nc  sfc_data.tile1.nc  sfc_data.tile3.nc  sfc_data.tile5.nc
```

And since both CDUMPs were produced using the same input data, the coldstart files are bit identical:

```
/scratch2/NCEPDEV/stmp1/George.Gayno/gdas.init.branch/output/gfs.20200101/06/model_data/atmos/input $ cmp sfc_data.tile6.nc /scratch2/NCEPDEV/stmp1/George.Gayno/gdas.init.branch/output/gdas.20200101/06/model_data/atmos/input/sfc_data.tile6.nc
/scratch2/NCEPDEV/stmp1/George.Gayno/gdas.init.branch/output/gfs.20200101/06/model_data/atmos/input
/scratch2/NCEPDEV/stmp1/George.Gayno/gdas.init.branch/output/gfs.20200101/06/model_data/atmos/input $ cmp gfs_data.tile3.nc /scratch2/NCEPDEV/stmp1/George.Gayno/gdas.init.branch/output/gdas.20200101/06/model_data/atmos/input/gfs_data.tile3.nc
/scratch2/NCEPDEV/stmp1/George.Gayno/gdas.init.branch/output/gfs.20200101/06/model_data/atmos/input
```

File: util/gdas_init/driver.s4.sh
Comment (GeorgeGayno-NOAA) [inline util/gdas_init/driver.s4.sh:1]: @DavidHuber-NOAA - Can you test the gdas_init utility on 's4' as well? You can copy the input data from Hera: /scratch2/NCEPDEV/stmp1/George.Gayno/gdas.init.branch/input.   In the config file, set yy=2020, mm=01, dd=01, hh=06.

Comment (DavidHuber-NOAA) [reply util/gdas_init/driver.s4.sh:1]: @GeorgeGayno-NOAA The gdas_init utility works on S4.  Thanks!

Comment (GeorgeGayno-NOAA) [reply util/gdas_init/driver.s4.sh:1]: Great!

**Comment by GeorgeGayno-NOAA** [comment]
@KateFriedman-NOAA or @aerorahul - Any further comments? If not, I would like to merge.

**Review by KateFriedman-NOAA — APPROVED** [review]
No further comments from me. Approving. Thanks @GeorgeGayno-NOAA !

**Comment by GeorgeGayno-NOAA** [comment]
> No further comments from me. Approving. Thanks @GeorgeGayno-NOAA !

Let me know if users complain that gdas_init no longer works with v14 or earlier data. If there are no complaints after six months or so, I will permanently remove support for the old data.

**Comment by KateFriedman-NOAA** [comment]
Will do, thanks @GeorgeGayno-NOAA !

**Pull request merged**